test/filter.test.ts

```diff
@@ -4,25 +4,31 @@ import test from 'ava';
 
 import * as promiseUtils from '../src/index';
 
+function delay(ms: number) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
 test('returns empty array when given no input', async t => {
   const output = await promiseUtils.filter(null as any, _.identity);
   t.deepEqual(output, []);
 });
 
-test('filters arrays', async t => {
-  const input = [1, 2];
+test('filters arrays and maintains order', async t => {
+  const input = [1, 2, 3, 4, 5, 6, 7, 8];
   const output = await promiseUtils.filter(input, async (value: any) => {
-    return value === 2;
+    await delay(_.random(5000)); // Add a variable delay to ensure we don't maintain order by luck
```

Comment:
not a big fan of adding a delay here since it makes the tests slower and is random so possible (though unlikely) to return in the same order... may create flakes.

Might be better to do something like this:
```
let nextRelease = input.length - 1;
const getNextRelease = () => nextRelease;
const output = await promiseUtils.filter(input, async (value, index) => {
  while (getNextRelease() !== index) {
    await promiseUtils.immediate();
  }
  nextRelease--;
  return value > 3;
});
```